src/Reflection/Adapter/ReflectionType.php

```diff
@@ -16,12 +18,20 @@ public function __construct(BetterReflectionType $betterReflectionType)
         $this->betterReflectionType = $betterReflectionType;
     }
 
-    public static function fromReturnTypeOrNull(?BetterReflectionType $betterReflectionType): ?self
+    public static function fromTypeOrNull(?BetterReflectionType $betterReflectionType): ReflectionUnionType|ReflectionNamedType|self|null
```

Comment:
We should be able to declare that if `T` (input) is `null`, then `null` is also the return value.

That way, hardcoded `ReflectionType::fromTypeOrNull(null)` calls should be detected as redundant